schematic/schemas/data_model_validator.py

```diff
@@ -10,8 +12,22 @@
 
 logger = logging.getLogger(__name__)
 
+# Characters display names of nodes that are not allowed
+BLACKLISTED_CHARACTERS = ["(", ")", ".", "-"]
+# Names of nodes that are used internally
+RESERVED_NAMES = {"entityId"}
 
-class DataModelValidator:
+
+NodeList = list[Tuple[Any, dict]]
+"""
+A list of node tuples.
+The first item is the name of the node.
+The second item is a dict of its fields.
+This object is gotten from doing nx.MultiDiGraph.nodes(data=True)
+"""
```

Comment:
Rather than `Tuple`, I much prefer the usage of classes to hold data. Maybe something like:

```
from dataclasses import dataclass
from typing import Any, Dict, List, Tuple

@dataclass
class Node:
    name: Any
    """Name of the node."""
    
    fields: Dict
    """Fields of the node"""
```

Then all the return types can be a List[Node]